cli/arguments/sketch.go

```diff
@@ -39,3 +42,53 @@ func InitSketchPath(sketchPath string) *paths.Path {
 	logrus.Infof("Reading sketch from dir: %s", wd)
 	return wd
 }
+
+// SketchPathCalculation returns the path and instance of the sketch and the discovery port.
+// Optionally, it warns the user when the sketch contains .pde files.
+func SketchPathCalculation(path string, port Port, instance *rpc.Instance, warn bool) (*paths.Path, *sketch.Sketch, *discovery.Port) {
+	sketchPath := InitSketch(path, warn)
+	sketch := CreateSketch(sketchPath)
+	discoveryPort := GetDiscoveryPort(instance, port, sketch)
+	return sketchPath, sketch, discoveryPort
+}
+
+// InitSketch is a helper function that determines the sketch path. Optionally, it warns the user when the sketch contains .pde files.
+func InitSketch(path string, warn bool) *paths.Path {
```

Comment:
```suggestion
func InitSketchPath(path string) *paths.Path {
```

We can always warn the users.